### ISSUE TITLE
Add a property to the views to indicate the scale

### ISSUE THREAD

**Issue by mattleibow** [mono/SkiaSharp#175]
Because the SkiaSharp views are using actual device pixels, this may be unexpected for some apps. I think we should expose two options:
1. A property to indicate the current scale
   This is especially noticeable on UWP when dragging the window from a hi-res display to a low res monitor, the image should stay the same size when drawing with the same coordinates.
2. A property to enable automatic scaling
   Currently, the rendering is done with device pixels, an option to draw with "base" pixels and automatically scale

These options might have a secondary option to indicate whether to scale the completed image (draw a bitmap of 100x100, but then scaling it up to 200x200 when blitting - may cause blurring) or to scale all drawing operations (drawing a bitmap of 200x200, but all draw operations are scaled up).

**Comment by jasonCodesAway** [comment]
Definitely looking forward to this feature. I couldn't figure out why my sketch wasn't taking up the full size of it's parent component. I think this feature will fix that for me. Thanks for the awesome code!

**Comment by mattleibow** [comment]
I want to look at this some more in the next release, but for this release I decided to expose the current canvas size as a property. Although we still can't control the scaling to be used, we can calculate the scaling:

```csharp
float scaling = view.CanvasSize.Width / view.Width;
```

This will get the canvas size from the last draw operation. If anything changes, such as the view's size, pixel density or scaling, then the canvas size will be recalculated on the next draw cycle. The update is not immediate. In the next release we could see about immediate updates, but for this release I am actually reading the values from the last canvas.

**Comment by mattleibow** [comment]
I have now also added a property (to the bitmap-based views) that can be used to disable automatic scaling:

    // assume view.Size == (100, 100)
    // assume iPhone retina (2x)

    // a larger canvas on hi-res displays
    view.IgnorePixelScaling = false;
    // view.CanvasSize == (200, 200)

    // the canvas size is directly related to display independent view size
    view.IgnorePixelScaling = true;
    // view.CanvasSize == (100, 100)

I only added this to raster-based views as this was a difference that might not be default for all platforms. GL-based views actually use the framebuffer dimensions and then is scaled to fit anyway. If this feature is desired on GL views for some reason, I can add it there. But right now, since scaling doesn't look so great, this is just for "compatibility" (if you can use that word).

**Comment by mattleibow** [comment]
@jasonCodesAway, remember you can also just scale your drawings up using the canvas matrix. In the latest released version (v1.55.0) I added a property called `CanvasSize` which you can use to detect the actual size of the canvas:

```csharp
var scale = view.CanvasSize.Width / view.Width;
```

This scale can also be used to scale up the canvas before drawing:

```csharp
SKCanvas canvas = ...;
canvas.Scale(scale);
```

In order to use other matrix translations, you can save and restore matrix state:

```csharp
canvas.Save();
// . . . 
canvas.Restore();
```

In the next version (probably v1.55.1) I added a property to disable screen scaling. But, this does not look so great as the image is just stretched. The best way to do this is to scale the canvas up, and draw to that. Then, when it is displayed, the image is still sharp. The reason this works is that it is just a matrix, and all drawing operations are scaled on the fly.

**Comment by jasonCodesAway** [comment]
Awesome! Thank you. I'm making good use of SkiaSharp in our Xamarin Forms app :) 👍
